src/components/link/link.component.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { assign } from 'lodash';
+import { Link as RouterLink } from 'react-router';
+import Icon from '../icon';
+import { validProps } from '../../utils/ether';
+import Event from '../../utils/helpers/events';
+import tagComponent from '../../utils/helpers/tags';
+import LinkStyle from './link.style';
+
+class Link extends React.Component {
+  static safeProps = ['onClick'];
+
+  constructor() {
+    super();
+    this.onKeyDown = this.onKeyDown.bind(this);
+  }
+
+  onKeyDown(ev) {
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(ev);
+    }
+
+    // return early if there is no onClick or there is a href prop
+    // or the event is not an enter key
+    if (this.props.href || !Event.isEnterKey(ev)) {
+      return;
+    }
+
+    if (this.props.onClick) {
+      this.props.onClick(ev);
+    }
+  }
+
+  get iconLeft() {
+    if (!this.props.icon || this.props.iconAlign !== 'left') {
+      return null;
+    }
+    return this.icon;
+  }
+
+  get iconRight() {
+    if (!this.props.icon || this.props.iconAlign !== 'right') {
+      return null;
+    }
+    return this.icon;
+  }
```

Comment:
you could use one method to render that icon:
```
renderLinkIcon = (currentAlignment) => {
  const hasProperAlignment = (this.props.icon && this.props.iconAlign === currentAlignment);

  return hasProperAlignment ? this.icon : null;
}
```

Reply:
`this.icon` return a html tag so we need to have both (right and left) functions to set the icon correctly,

Reply:
you use that method with an argument: `renderLinkIcon('left')`